lib/tests/unit/request_processor_test.cc

```diff
@@ -20,132 +20,112 @@
 namespace PXPAgent {
 
 namespace lth_jc = leatherman::json_container;
-
-static const std::string TEST_BROKER_WS_URI { "wss://127.0.0.1:8090/pxp/" };
-static const std::string CA { getCaPath() };
-static const std::string CERT { getCertPath() };
-static const std::string KEY { getKeyPath() };
-static const std::string MODULES { PXP_AGENT_ROOT_PATH
-                            + std::string { "/lib/tests/resources/modules" } };
-static const std::string SPOOL { PXP_AGENT_ROOT_PATH
-                          + std::string { "/lib/tests/resources/tmp/" } };
-
-static const Configuration::Agent agent_configuration { MODULES,
-                                                        TEST_BROKER_WS_URI,
-                                                        CA,
-                                                        CERT,
-                                                        KEY,
-                                                        SPOOL,
-                                                        "",  // modules config dir
-                                                        "test_agent",
-                                                        5000 };
+namespace pcp_util = PCPClient::Util;
+namespace fs = boost::filesystem;
 
 TEST_CASE("RequestProcessor::RequestProcessor", "[agent]") {
-    auto c_ptr = std::make_shared<PXPConnector>(agent_configuration);
+    auto c_ptr = std::make_shared<PXPConnector>(AGENT_CONFIGURATION);
 
     SECTION("successfully instantiates with valid arguments") {
-        REQUIRE_NOTHROW(RequestProcessor(c_ptr, agent_configuration));
+        REQUIRE_NOTHROW(RequestProcessor(c_ptr, AGENT_CONFIGURATION));
     };
 
     SECTION("instantiates if the specified modules directory does not exist") {
-        Configuration::Agent a_c  = agent_configuration;
+        Configuration::Agent a_c  = AGENT_CONFIGURATION;
         a_c.modules_dir += "/fake_dir";
 
         REQUIRE_NOTHROW(RequestProcessor(c_ptr, a_c));
     };
 
-    boost::filesystem::remove_all(SPOOL);
+    fs::remove_all(SPOOL);
 }
 
-#ifdef TEST_VIRTUAL
+TEST_CASE("RequestProcessor::hasModule", "[agent]") {
+    AGENT_CONFIGURATION.modules_config_dir = VALID_MODULES_CONFIG;
```

Comment:
We should either create a copy of the configuration, or make sure the ```AGENT_CONFIGURATION.modules_config_dir``` is reset after the test case.